### ISSUE TITLE
helm.go:71: error loading config file permission denied

### ISSUE THREAD

**Issue by IainColledge** [helm/helm#9683]
I have Ubuntu 20.04.02 with microk8s 1.20.6 installed along with helm3 3.1.2

The following is enabled on the cluster:

dashboard       # The Kubernetes dashboard
dns                  # CoreDNS
ha-cluster        # Configure high availability on the current node
helm3              # Helm 3 - Kubernetes package manager
ingress             # Ingress controller for external access
metrics-server # K8s Metrics Server for API access to service metrics
rbac                 # Role-Based Access Control for authorisation
registry            # Private image registry exposed on localhost:32000
storage            # Storage class; allocates storage from host directory

To overcome that helm was reporting kubernetes was unreachable I did the following:

microk8s.kubectl config view --raw > $HOME/.kube/config

I had to put the following in .profile for helm to look for the file

export KUBECONFIG=$HOME/.kube/config 

The error message I have when trying to run say helm3 list is:

helm.go:71: error loading config file "/home/iain/.kube/config": open /home/iain/.kube/config: permission denied

Permissions for .kube I have set to:

drwxrwxrwx 3 iain iain 4096 May  9 09:11 .kube

Permissions for .kube/config I have set to:

-rwxrwxrwx 1 iain iain 1842 May  9 10:03 config

I can view the config file with cat no problem, please can someone help me with this problem

Output of `helm version`:

I have to remove KUBECONFIG to see:

version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}

Output of `kubectl version`:

Client Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.6-34+e4abae43f6acde", GitCommit:"e4abae43f6acde18ad466c0d3f7e22a32a388229", GitTreeState:"clean", BuildDate:"2021-04-17T11:03:34Z", GoVersion:"go1.15.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.6-34+e4abae43f6acde", GitCommit:"e4abae43f6acde18ad466c0d3f7e22a32a388229", GitTreeState:"clean", BuildDate:"2021-04-17T11:04:57Z", GoVersion:"go1.15.10", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

Microk8s 3 node cluster

**Comment by IainColledge** [comment]
Using the helm CLI installed using chocolatey from a remote windows 10 machine using an identical config file works.

**Comment by bacongobbler** [comment]
Interesting. The only cause I've heard this error occur would be due to bad file permissions for your `~/.kube/config`. Let us know if you can determine anything.

**Comment by IainColledge** [comment]
Yeah I know, it's very strange and you can see the permissions are wide open.

All I can think of is the snap version is very old with this version of Ubuntu: v3.1.2 given that the windows 10 version is v3.5.4 and at some time in the future a new package might be put in snap and everything will align to make it automagically work again.

**Comment by IainColledge** [comment]
Good points, I'll try a wipe and reinstall and comment with the result, even if it ends up being between keyboard and chair. :)

Thanks for your help.

**Comment by IainColledge** [comment]
Ok, so if one types helm3 at the command line in ubuntu you'll get instructed:

Command 'helm3' not found, but can be installed with:

sudo snap install helm3

So that is why I installed it that way, going to the helm documentation and installing with:

sudo snap install helm --classic

will install the right version and it works as expected.
